site/src/main/paradox/io/Parquet.md

```diff
@@ -112,6 +112,66 @@ def yourAvroSchema: org.apache.avro.Schema = ???
 def result = input.saveAsParquetAvroFile("gs://path-to-data/lake/output", schema = yourAvroSchema)
 ```
 
+### Logical Types
+
+If your Avro schema contains a logical type, you'll need to supply an additional Configuration parameter for your reads and writes.
+
+If you're using the default version of Avro (1.8), you can use Scio's pre-built logical type conversions:
+
+```scala mdoc:reset
+import com.spotify.scio._
+import com.spotify.scio.values.SCollection
+import com.spotify.scio.parquet.avro._
+import com.spotify.scio.avro.TestRecord
+
+val sc: ScioContext = ScioContext()
+val data: SCollection[TestRecord] = sc.parallelize(List[TestRecord]())
+
+// Reads
+import com.spotify.scio.parquet.ParquetConfiguration
+
+import org.apache.parquet.avro.AvroReadSupport
+
+sc.parquetAvroFile(
+  "somePath",
+  conf = ParquetConfiguration.of(AvroReadSupport.AVRO_DATA_SUPPLIER -> classOf[LogicalTypeSupplier])
+)
+
+// Writes
+import org.apache.parquet.avro.AvroWriteSupport
+
+data.saveAsParquetAvroFile(
+  "somePath",
+  conf = ParquetConfiguration.of(AvroWriteSupport.AVRO_DATA_SUPPLIER -> classOf[LogicalTypeSupplier])
+)
+```
+
+(If you're using `scio-smb`, you can use the provided class `org.apache.beam.sdk.extensions.smb.AvroLogicalTypeSupplier` instead.)
+
+If you're using Avro 1.11, you'll have to create your own logical type supplier class, as Scio's `LogicalTypeSupplier` uses
+classes present in Avro 1.8 but not 1.11. A sample Avro 1.11 logical-type supplier might look like:
+
+```scala
+import org.apache.avro.Conversions;
+import org.apache.avro.data.TimeConversions;
+import org.apache.avro.generic.GenericData;
+import org.apache.avro.specific.SpecificData;
+import org.apache.parquet.avro.AvroDataSupplier;
+
+case class AvroLogicalTypeSupplier() extends AvroDataSupplier {
+  override def get(): GenericData = {
+    val specificData = SpecificData.get()
+
+    // Add conversions as needed
+    specificData.addLogicalTypeConversion(new TimeConversions.TimestampMillisConversion())
+
+    specificData
+  }
+}
```

Comment:
I feel we should be able to do better that this.
With specific records, all conversions are added to a pre-generated model contained in the static `$MODEL` field of the class, and can be accessed with `SpecificData.getForClass` in avro 1.9+. It is still possible to get this work with avro 1.8 with some tricks.
This supplier should only be required for records generated with `ReflectData` or `GeneridData` where conversions must be explicitly given.

Reply:
I did see `$MODEL` -- as well as a static field `org.apache.avro.Conversion<?>[] conversions` that has everything we need -- but was hesitant to use them because in Avro 1.8 at least, they're both private so I'd have to use reflection on private/maybe unstable fields. It would be ideal to get the conversions directly from the specific record class though

Reply:
Looked into this a bit more!

I think you're right, we can definitely go from `T`/`Schema` --> a `SpecificData` model populated with conversion types by re-using Avro's model-cache logic.

But the complex bit is actually wiring this model in. Parquet allows you to specify a model by either (a) setting `.withDataModel()` on the read/write builder, or (b) supplying the Configuration `*DATA_SUPPLIER` param. To use our derived `SpecificData` model, we'd have to use `.withDataModel` directly, I think -- I can't see how we'd make this work with a [generic Supplier class](https://github.com/apache/parquet-mr/blob/master/parquet-avro/src/main/java/org/apache/parquet/avro/AvroDataSupplier.java) that requires a no-arg constructor.

For Parquet writes, we do invoke the write builder directly, so we could take our derived `SpecificData` and set it using [.withDataModel](https://github.com/spotify/scio/blob/main/scio-parquet/src/main/java/com/spotify/scio/parquet/avro/ParquetAvroFileBasedSink.java#L127).

But for reads, we don't get to directly access the `AvroParquetReader` and set `.withDataModel`--there are too many layers of indirection--so the only way we can specify a data model is to configure the `DATA_SUPPLIER` parameter with the name of a Supplier class, which as I mentioned above, I don't know how we could wire our custom data model into.